### PR TITLE
Replaced GetAgentInfo => GetAgent

### PR DESCRIPTION
**Pull request by mingazhev**

### CODE REVIEW

File: Api/Controllers/AgentsController.cs [hunk 1]
```diff
@@ -26,6 +26,7 @@ namespace HappyTravel.Edo.Api.Controllers
     public class AgentsController : ControllerBase
     {
         public AgentsController(IAgentRegistrationService agentRegistrationService, IAgentContext agentContext,
+            IAgentContextInternal agentContextInternal,
```
Comment (kirillta) [inline Api/Controllers/AgentsController.cs:29]: This internal context thing confuses me. Why do we separate them. Also `GetAgentInfo` and `SetAgentInfo` lays in different classes

Comment (mingazhev) [reply Api/Controllers/AgentsController.cs:29]: Internal context is used mostly for internal use and can safely return failing result.
SetAgentInfo will soon be removed

File: Api/Controllers/LocationsController.cs [hunk 2]
```diff
@@ -48,11 +49,10 @@ public LocationsController(IAgentContext agentContext, ILocationService service,
         [HttpGet("predictions")]
         [ProducesResponseType(typeof(List<Prediction>), (int) HttpStatusCode.OK)]
         [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.BadRequest)]
+        [AgentRequired]
         public async Task<IActionResult> GetLocationPredictions([FromQuery] string query, [FromQuery] [Required] string sessionId)
         {
-            var (_, isAgentFailure, agentInfo, agentError) = await _agentContext.GetAgentInfo();
-            if (isAgentFailure)
-                return BadRequest(ProblemDetailsBuilder.Build(agentError));
+            var agentInfo = await _agentContext.GetAgent();
```
[the remark this answers is not in the document]
Comment (mingazhev) [reply Api/Controllers/LocationsController.cs:55]: Fixed

File: Api/Services/Accommodations/Bookings/BookingRecordsManager.cs [hunk 3]
```diff
@@ -36,14 +36,10 @@ internal class BookingRecordsManager : IBookingRecordsManager
...
-        public async Task<Result<string>> Register(AccommodationBookingRequest bookingRequest, BookingAvailabilityInfo availabilityInfo, string languageCode)
+        public async Task<string> Register(AccommodationBookingRequest bookingRequest, BookingAvailabilityInfo availabilityInfo, string languageCode)
         {
-            var (_, isAgentFailure, agentInfo, agentError) = await _agentContext.GetAgentInfo();
-
-            return isAgentFailure
-                ? ProblemDetailsBuilder.Fail<string>(agentError)
-                : Result.Ok(await CreateBooking());
-
+            var agentInfo = await _agentContext.GetAgent();
```
[the remark this answers is not in the document]
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/BookingRecordsManager.cs:41]: Fixed

File: Api/Services/Agents/CounterpartyService.cs [hunk 4]
```diff
@@ -146,10 +146,7 @@ public Task<Result<Agency>> AddAgency(int counterpartyId, AgencyInfo agency)
 
             async Task<bool> HasPermissions()
             {
-                var (_, isFailure, agentInfo, _) = await _agentContext.GetAgentInfo();
-                if (isFailure)
-                    return false;
-
+                var agentInfo = await _agentContext.GetAgent();
```
[the remark this answers is not in the document]
Comment (mingazhev) [reply Api/Services/Agents/CounterpartyService.cs:149]: Fixed

File: Api/Controllers/AgentsController.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Controllers/AgentsController.cs:29]: About the reason to separate. There are two similar methods with similar meaning - one throws exception when fails and another just returns Fail result. I thought that having these both methods in a single interface is more confusing.

**Pull request merged**